### ISSUE TITLE
Improve Progressive Web App

### ISSUE THREAD

**Issue by darkf3d3** [PrestaShop/PrestaShop#10366]
Prestashop 1.7.4.2 demo score only 42/100 on Progressive Web App using Google Lighthouse tool.
That would be a good thing to improve this.

**Comment by khouloudbelguith** [comment]
Hi @darfk3d3,

Thanks for your report.
I manage to reproduce the issue with the last version 1.7.4.2.
https://drive.google.com/file/d/1I0NbBg-bwYZtM_unXuTczN_a_Z0Q6Sgz/view

Best regards, Khouloud

**Comment by rdy4ever** [comment]
Well, Prestashop is not by default a PWA. So maybe you should install a PWA module from the [add-ons store](https://addons.prestashop.com/en/search?search_query=pwa). Or, implement something like [this](https://github.com/DivanteLtd/vue-storefront) if you have the skills. Plese note that Magento also scores just 42 in Lighthouse in its default configuration. However, with Vue Storefront implemented, it scores 100:
![image](https://user-images.githubusercontent.com/12268701/45311110-d4ea9080-b530-11e8-8676-4ffce4d4935d.png)

**Comment by darkf3d3** [comment]
I will look for a developer that it's able to implemente Wue Storefront because Prestashop module seem not working properly, because on their front end demo they score no more than default PS demo.
A part from this I think that it should be implemented into next PS releases.

**Comment by ndkdesign** [comment]
this one will do the job: https://addons.prestashop.com/fr/mobile/31205-pwa-notification-push.html

**Comment by darkf3d3** [comment]
I saw your module, but the demo scored only 38/100 on Progressive Web App, that's a lower score than Prestashop without the module.

**Comment by ndkdesign** [comment]
hello test here please
https://shop.ndk-design.fr/ <https://shop.ndk-design.fr/>

demo site has to be debugged

hendrik masson
postmaster@ndk-design.fr

**Comment by darkf3d3** [comment]
Thank you hendrik, I checked that page and it's score it's higher (69/100), but there's still 2 points not passed:
1. Does not respond with a 200 when offline
2. User will not be prompted to Install the Web App
Failures: Service worker does not successfully serve the manifest's start_url, Timed out waiting for fetched start_url.

With your module also that 2 points can be fixed to obtain 100/100 on Lighthouse PWA?

**Comment by ndkdesign** [comment]
You're right but point 2 works, try on mobile Le 12 sept. 2018 10:48 AM, darfk3d3 <notifications@github.com> a écrit :Thank you hendrik, I checked that page and it's score it's higher (69/100), but there's still 2 points not passed:
Does not respond with a 200 when offlineUser will not be prompted to Install the Web App
Failures: Service worker does not successfully serve the manifest's start_url, Timed out waiting for fetched start_url.
With your module also that 2 points can be fixed to obtain 100/100 on Lighthouse PWA?

**Comment by ndkdesign** [comment]
on mobile you will be able to see offline page already viewed...
hendrik masson
postmaster@ndk-design.fr

**Comment by darkf3d3** [comment]
But that's strange that google show alert for them, because when testing the website with Lighthouse, google test it also simulating a mobile device.

**Comment by ndkdesign** [comment]
hello you can retry now score is 100

hendrik masson
postmaster@ndk-design.fr

**Comment by marionf** [comment]
@darfk3d3 @ndkdesign 
We can close this issue if a module meets your request ?

**Comment by darkf3d3** [comment]
Most of the things missing on Prestashop can be added with a module, but I think that this is not a reason to doesn't develop a new feature for Prestashop. Otherwise no new features will be developed by Prestashop team only because often there's a module that do this.
PWA should be a native feature, not having it's a lack of Prestashop and should be included into PS core.

**Comment by marionf** [comment]
@colinegin Do you think this should be include as a new feature ?

**Comment by darkf3d3** [comment]
EDIT: Wrong post

**Comment by colinegin** [comment]
Hello,

Thanks a lot for your feedback, and please excuse the delay.

This project is on our roadmap, you can follow our progress on the public roadmap available on Github.
In the meantime, I put this issue on "To do" with a "PWA" label, we've share it on the PrestaShop uservoice, you can vote for the idea here : http://feedback.prestashop.com/forums/387864-prestashop-1-7-x. We prioritize projects on our roadmap according to ideas which have been voted the most.

Do not hesitate if you have more ideas to share !

Have a nice day,

Coline
